Source/JSONAPISerialization.swift

```diff
@@ -0,0 +1,206 @@
+   4. `relationships`: all the properties that conform to JSONAPIEntity or array of JSONAPIEntity are recognized as relationships.
+ 
+ * Note: When `JSONAPIEntity` is serialized as relationship only `id` and `type` will be included.
+ 
+ * [See the JSON API documentation](http://jsonapi.org/format/#document-resource-objects)
+ */
+public protocol JSONAPIEntity: CustomSerializable, JSONAPISerializable {
+    /// The type of this entity, by default the lowercase class name is used.
+    var type: String { get }
+    /// The id of the entity
+    var id: String { get }
+}
+
+/**
+ *  An object responsible to serialize a `JSONAPIEntity` or an array of `JSONAPIEntity` conforming to JSON API
+ */
+public struct JSONAPISerializer<T: JSONAPIEntity>: CustomSerializable {
+    
+    private let data: AnyObject
+    
+    /**
+     Initialize a serializer with a single `JSONAPIEntity`
+     
+     - parameter object: A `JSONAPIEntities`
+     
+     - returns: A serializable object that serializes a `JSONAPIEntity` conforming to JSON API
+     */
+    public init(_ object: T) {
+        data = object.serialize()! // can't fail, JSONAPIEntity must always be serializable
+    }
+    
+    /**
+     Initialize a serializer with an array of `JSONAPIEntity`
+     
+     - parameter objects: An array of `JSONAPIEntity`
+     
+     - returns: A serializable object that serializes an array of `JSONAPIEntity` conforming to JSON API
+     */
+    public init(_ objects: [T]) {
+        data = objects.serialize()! // can't fail, JSONAPIEntity must always be serializable
+    }
+    
+    // MARK: CustomSerializable
+
+    public func customSerialize() -> AnyObject? {
+        return ["data": data]
+    }
+}
+
+// MARK: - Extensions
+
+extension Array: JSONAPISerializable {
+    
+    // MARK: JSONAPISerializable
+    
+    public func data(includeRelationships includeRelationships: Bool, includeAttributes: Bool) -> AnyObject? {
+        return Element.self is JSONAPISerializable.Type ? flatMap { ($0 as? JSONAPISerializable)?.data(includeRelationships: includeRelationships, includeAttributes: includeAttributes) } : nil
+    }
+}
+
+extension PropertyPolicy: JSONAPISerializable {
+
+    // MARK: JSONAPISerializable
+    
+    public func data(includeRelationships includeRelationships: Bool, includeAttributes: Bool) -> AnyObject? {
+        guard Wrapped.self is JSONAPISerializable.Type else {
+            return nil
+        }
+        
+        switch self {
+        case let .Some(value):
+            if let value = value as? JSONAPISerializable {
+                return value.data(includeRelationships: includeRelationships, includeAttributes: includeAttributes)
+            }
+            
+        case .Null:
+            return [String: AnyObject]() // included as relationship but empty
+        case .None:
+            return nil
+        }
```

Comment:
`Null` returns empty dict and `None` nil, a bit confusing for me after reading `PropertyPolicy` docs